satpy/tests/writer_tests/test_mitiff.py

```diff
@@ -557,16 +562,18 @@ def test_save_datasets(self):
         w.save_datasets(dataset)
         filename = (dataset[0].attrs['metadata_requirements']['file_pattern']).format(
             start_time=dataset[0].attrs['start_time'])
-        tif = TIFF.open(os.path.join(self.base_dir, filename))
-        for image in tif.iter_images():
-            np.testing.assert_allclose(image, expected, atol=1.e-6, rtol=0)
+        pillow_tif = Image.open(os.path.join(self.base_dir, filename))
+        for frame_no in range(pillow_tif.n_frames):
+            pillow_tif.seek(frame_no)
+            np.testing.assert_allclose(np.asarray(pillow_tif.getdata()).reshape((100, 200)),
+                                       expected, atol=1.e-6, rtol=0)
```

Comment:
could this be extracted as a method of the test class, that way the chunk on lines 585-589 could be removed also?

Reply:
ok